src/docs/truffle/getting-started/installation.md

```diff
@@ -4,17 +4,44 @@ layout: docs.hbs
 ---
 # Installation
 
-```bash
-npm install -g truffle
-```
-
 ## Requirements
 
-* NodeJS v12 or later
-* Windows, Linux or Mac OS X
+* [NodeJS v12, or later](#install-nodejs)
+* Windows, Linux, or macOS
+
+## Install NodeJS
+
+### Linux and macOS
+
+NPM recommends [installing Node.js with a version manager (NVM)](https://npm.github.io/installation-setup-docs/installing/using-a-node-version-manager.html)
+to avoid permission errors when installing globally.
+
+**Important**: Truffle requires `node-gyp` for compiling native addon modules for Node.js.  Node-gyp in
+turn has [dependencies](https://github.com/nodejs/node-gyp) which may be missing, causing the Truffle install to fail.
+Truffle recommends installing the following to avoid getting dependency errors:
+
+* macOS only: `xcode-select --install`
+* Linux (Ubuntu) only: `sudo apt install make g++`
 
-Truffle also requires that you have a running Ethereum client which supports the standard JSON RPC API (which is nearly all of them). There are many to choose from, and some better than others for development. We'll discuss them in detail in the [Choosing an Ethereum client](/docs/truffle/reference/choosing-an-ethereum-client) section.
+### Windows
+
+Truffle recommends using the installer available from the [nodejs site](https://nodejs.org/en/download/).
+
+Ensure you select **Automatically install the necessary tools...** during the install to install the
+required Visual Studio build tools, Python, and Chocolately package manager.
+
+![Architecture](/img/docs/truffle/installation/windows-nodejs.png)
+
+## Install Truffle
```

Comment:
Not sure what best practices for docs are, but would it be worthwhile to include a test see that truffle is installed by using truffle version?

Unfortunately, truffle spits out a long laundry list installation warnings, which a lot of users get confused by and think installation was unsuccessful because of it

Reply:
Fixed